src/test/java/com/microsoft/sqlserver/jdbc/connection/TimeoutTest.java

```diff
@@ -154,6 +155,33 @@ public void testConnectRetryServerError() {
         verifyTimeout(timerEnd - timerStart, defaultTimeout);
     }
 
+    // Test connect retry for database error using Datasource
+    @Test
+    public void testConnectRetryServerErrorDS() {
+        long timerEnd = 0;
+        long timerStart = System.currentTimeMillis();
+
+        // non existent database with interval < loginTimeout this will generate a 4060 transient error and retry
+        int connectRetryCount = new Random().nextInt(255);
```

Comment:
Random().nextInt(255) excludes 255 so this value won't be tested. I think we need to use Random().nextInt(256) here.